### ISSUE TITLE
Cannot find module '@t3-oss/env-core' or its corresponding type declarations.

### ISSUE THREAD

**Issue by ericmesmo** [t3-oss/t3-env#245]
I'm using turborepo and in it I have a package to get the application's variables, whether front or back. 

When using the @repo/env-config package within the api application, which uses nestjs, an error is returned: Cannot find module '@t3-oss/env-core' or its corresponding type declarations.

I couldn't find a solution, could someone help me?

![image](https://github.com/t3-oss/t3-env/assets/24236505/3e5f9573-4f27-4fdf-aeab-788c332d8d76)

A repo to reproduce the error: https://github.com/ericmesmo/reproducible

**Comment by kon-pas** [comment]
Resolved here: #189 

@ericmesmo In your case, to use a distinct `tsconfig` in a package, it has to go through the dev or build step.

First, configure your dist directory in `env-config/tsconfig.json`:

```diff
{
  "extends": "@repo/typescript-config/base.json",
  "include": ["index.ts"],
+ "exclude": ["dist"],
+ "compilerOptions": {
+   "outDir": "dist"
+ },
}
```

Then, update endpoints and scripts in `env-config/package.json`:

```diff
{
  "name": "@repo/env-config",
  "version": "1.0.0",
  "description": "",
  "private": true,
- "main": "index.ts",
+ "main": "dist/index.js",
- "types": "index.ts",
+ "types": "dist/index.d.ts",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
+   "build": "tsc",
+   "dev": "tsc --watch"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@t3-oss/env-core": "^0.10.1",
    "ts-node": "^10.9.1",
    "tsup": "^8.1.0",
    "typescript": "^5.4.5",
    "zod": "^3.23.8"
  },
  "devDependencies": {
    "@types/node": "^20.14.2",
    "typescript": "^5.4.5",
    "@repo/typescript-config": "workspace:*"
  }
}
```

Now you can build your apps that use the `env-config` package:

```sh
pnpm build
```

**Comment by juliusmarminge** [comment]
> I couldn't find a solution, could someone help me?

The error explains how to solve this: 
![CleanShot 2024-07-22 at 10 06 09](https://github.com/user-attachments/assets/c45d16e3-0826-4475-a77f-779a99a1aa99)

This is also mentioned in the readme of this repo and in the docs
